bridge/slack/helpers.go

```diff
@@ -106,6 +107,65 @@ func (b *Bslack) populateChannels() {
 	b.channelsByName = newChannelsByName
 }
 
+// populateReceivedMessage shapes the initial Matterbridge message that we will forward to the
+// router before we apply message-dependent modifications.
+func (b *Bslack) populateReceivedMessage(ev *slack.MessageEvent) (*config.Message, error) {
+	// Use our own func because rtm.GetChannelInfo doesn't work for private channels.
+	channel, err := b.getChannelByID(ev.Channel)
+	if err != nil {
+		return nil, err
+	}
+
+	rmsg := &config.Message{
+		Text:    ev.Text,
+		Channel: channel.Name,
+		Account: b.Account,
+		ID:      "slack " + ev.Timestamp,
+		Extra:   make(map[string][]interface{}),
+	}
+	if b.useChannelID {
+		rmsg.Channel = "ID:" + channel.ID
+	}
+
+	if err = b.populateMessageWithUserInfo(ev, rmsg); err != nil {
+		return nil, err
+	}
+	return rmsg, err
+}
+
+func (b *Bslack) populateMessageWithUserInfo(ev *slack.MessageEvent, rmsg *config.Message) error {
+	if ev.SubType == sMessageDeleted || ev.SubType == sFileComment {
+		return nil
+	}
+
+	if ev.BotID != "" && b.GetString(tokenConfig) != "" {
+		bot, err := b.rtm.GetBotInfo(ev.BotID)
+		if err != nil {
+			return err
+		}
+		if bot.Name != "" && bot.Name != "Slack API Tester" {
+			rmsg.Username = bot.Name
+			if ev.Username != "" {
+				rmsg.Username = ev.Username
+			}
+			rmsg.UserID = bot.ID
+		}
+	}
+
+	if rmsg.Username == "" && ev.User != "" {
```

Comment:
same comment here as above, I'm not 100% sure anymore but there was a reason :)

Reply:
I am trying to find a possible explanation but I really can't come up with anything. According to the old code's comment:
```go
  // when using webhookURL we can't check if it's our webhook or not for now
```

However there is no later point either where we check this either so I don't understand the "not for now" part.

That said according to https://api.slack.com/custom-integrations/outgoing-webhooks we should definitely be able to verify the user's identity as the `ev.Username` field will be set instead of the `ev.User` one. I've updated the accordingly.

Reply:
Seems to be added in commit bd97357f8d30322dc9c9745c47b9461d862844ba
IIRC, the problem is that if we see messages in slack where `ev.BotID` is set and we also send messages to that channel via webhooks, we couldn't figure out if it was our own webhook that send the message or if it was another bot. So we just skip those message from other bots if we're using webhooks ourselves.